### PR TITLE
LESC/Legacy Security passkey support

### PR DESCRIPTION
**Pull request by estutzenberger**
* Implement support for passkey based LESC/Legacy pairing
* Update library to use go modules
* Update namespace to use rigado instead of go-ble
* Add support for the central to provide a GATT server to connected peripherals
* Add smp heart rate collection example

### CODE REVIEW

File: linux/hci/smp/handler.go [hunk 1]
```diff
@@ -215,6 +250,87 @@ func smpOnMasterIdentification(t *transport, in pdu) ([]byte, error) {
...
+		return Oob
+	}
+
+	if req.AuthReq & mitmMask == 0x00 &&
+		rsp.AuthReq & mitmMask == 0x00 {
+		return JustWorks
+	}
+
+	if t.pairing.legacy {
+		return ioCapsTableLegacy[rsp.IoCap][req.IoCap]
```
Comment (donaldschen) [inline linux/hci/smp/handler.go]: I would either bounds check these indexes or change to a map lookup

File: linux/hci/smp/handler.go [hunk 2]
```diff
@@ -215,6 +250,87 @@ func smpOnMasterIdentification(t *transport, in pdu) ([]byte, error) {
...
+	}
+
+	return ioCapsTableSC[rsp.IoCap][req.IoCap]
```
Comment (donaldschen) [inline linux/hci/smp/handler.go]: same as above

File: linux/hci/smp/manager.go [hunk 3]
```diff
@@ -93,8 +106,32 @@ func (m *manager) Bond() error {
 	//todo: can this be made less bad??
 	m.pairing.scECDHKeys = keys
 	m.t.pairing = m.pairing
+	m.t.pairing.authData = authData
+
+	//set a default timeout
+	if to == time.Duration(0) {
```
Comment (donaldschen) [inline linux/hci/smp/manager.go]: `to<=0`

File: linux/hci/smp/handler.go (hunk 1 above)
Comment (estutzenberger) [reply linux/hci/smp/handler.go]: Fixed

File: linux/hci/smp/handler.go (hunk 2 above)
Comment (estutzenberger) [reply linux/hci/smp/handler.go]: Fixed

File: linux/hci/smp/manager.go (hunk 3 above)
Comment (estutzenberger) [reply linux/hci/smp/manager.go]: Fixed

**Pull request merged**